src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java

```diff
@@ -88,53 +93,74 @@ public String getTransactionIdForConsentRequest(String consentRequestId) {
                 .flatMap(dataFlowRepository::getTransactionId).block();
     }
 
-    public Flux<PatientHealthInfoStatus> fetchHealthInformationStatus(List<String> dataRequestIds, String requesterId) {
-        return patientConsentRepository.fetchConsentRequestIds(dataRequestIds)
+    public Flux<PatientHealthInfoStatus> fetchHealthInformationStatus(List<String> dataRequestIds) {
+        var dataReqUUIDs = dataRequestIds.stream().filter(this::isUUID).collect(Collectors.toSet());
+        return dataFlowRepository.fetchPatientDataRequestDetails(dataReqUUIDs)
                 .collectList()
-                .flatMapMany(requestMappings -> {
+                .flatMapMany(patientDataRequestDetails -> {
+                    var detailsByDataReqId = patientDataRequestDetails.stream()
+                            .collect(Collectors.groupingBy(PatientDataRequestDetail::getDataRequestId));
+
                     var patientHealthInfoStatuses = new ArrayList<PatientHealthInfoStatus>();
-                    var statusesByConsentReqId = new HashMap<String, PatientHealthInfoStatus>();
-
-                    requestMappings.forEach(requestMapping -> {
-                        var patientHealthInfoStatus = PatientHealthInfoStatus.builder()
-                                .hipId(requestMapping.getHipId())
-                                .requestId(requestMapping.getDataRequestId())
-                                .status(DataRequestStatus.PROCESSING).build();
-                        if (StringUtils.isEmpty(requestMapping.getConsentRequestId())) {
-                            patientHealthInfoStatuses.add(patientHealthInfoStatus);
+                    detailsByDataReqId.forEach((dataReqId, dataRequestDetails) -> {
+                        var dataRequestDetail = dataRequestDetails.get(0);
```

Comment:
why get by index? also is it guaranteed to send over only 1 entry? what if empty or more than 1?

Reply:
Yeah, it is guaranteed that there will be at least one entry.